addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/handler/LGWebOSHandler.java

```diff
@@ -0,0 +1,223 @@
+
+/**
+ * The {@link LGWebOSHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sebastian Prehn - Initial contribution
+ */
+public class LGWebOSHandler extends BaseThingHandler implements ConnectableDeviceListener, DiscoveryManagerListener {
+
+    private Logger logger = LoggerFactory.getLogger(LGWebOSHandler.class);
+    private DiscoveryManager discoveryManager;
+
+    // ChannelID to CommandHandler Map
+    private final Map<String, ChannelHandler> channelHandlers = ImmutableMap.<String, ChannelHandler> builder()
+            .put(CHANNEL_VOLUME, new VolumeControlVolume()).put(CHANNEL_POWER, new PowerControlPower())
+            .put(CHANNEL_MUTE, new VolumeControlMute()).put(CHANNEL_CHANNEL, new TVControlChannel())
+            .put(CHANNEL_CHANNEL_UP, new TVControlUp()).put(CHANNEL_CHANNEL_DOWN, new TVControlDown())
+            .put(CHANNEL_CHANNEL_NAME, new TVControlChannelName()).put(CHANNEL_APP_LAUNCHER, new LauncherApplication())
+            .put(CHANNEL_MEDIA_STOP, new MediaControlStop()).put(CHANNEL_TOAST, new ToastControlToast())
+            .put(CHANNEL_MEDIA_PLAYER, new MediaControlPlayer()).build();
+
+    public LGWebOSHandler(Thing thing, DiscoveryManager discoveryManager) {
+        super(thing);
+        this.discoveryManager = discoveryManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("internalReceiveCommand({},{}) is called", channelUID, command);
+        ChannelHandler handler = channelHandlers.get(channelUID.getId());
+        if (handler == null) {
+            logger.warn(
+                    "Unable to handle command {}. No handler found for channel {}. This must not happen. Please report as a bug.",
+                    command, channelUID);
+            return;
+        }
+        final ConnectableDevice device = getDevice();
+        if (device == null) {
+            logger.warn(
+                    "Unable to handle command {} for channel {}. Device {} not found. This should not happen at this point.",
+                    command, channelUID, getThing().getUID());
+            return;
+        }
+        handler.onReceiveCommand(device, command);
+    }
+
+    private ConnectableDevice getDevice() {
+        String ip = this.getThing().getProperties().get(PROPERTY_IP_ADDRESS);
+        return this.discoveryManager.getCompatibleDevices().get(ip);
+    }
+
+    @Override
+    public void initialize() {
+        this.discoveryManager.addListener(this);
+
+        ConnectableDevice device = getDevice();
+        if (device == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "TV is off");
+        } else {
+            device.addListener(this);
+            if (device.isConnected()) {
+                updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "Connected");
+            } else {
+                device.connect();
```

Comment:
I'll add updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "Connecting ..."); 

on success onDeviceReady will be called,if pairing is required onPairingRequired, otherwise onConnectionFailed
all set the appropriate status

Reply:
I have also added a comment into the code similar to the above